### PR TITLE
fix: afterRenderEvent doesn't work in readonly mode ( fix #1327 )

### LINKED ISSUE #1327: afterRenderEvent not fired when in readonly mode
## Version
TOAST UI Calendar 2nd Edition @version 2.1.3 

## Test Environment

Browser: Chrome Version 108.0.5359.99 (Official Build) (64-bit)
## Current Behavior
When using the calendar in readonly mode the ```afterRenderEvent``` is not fired when redering events, this is the code i use:

```js
const app = calendar.querySelector(".tui-main-calendar");

calendar.tuiCalendar = new tui.Calendar(app, {
    defaultView: 'month',
    disableDblClick: true,
    usageStatistics: false,
    useDetailPopup: false,
    useFormPopup: false,
    isReadOnly: true,
    month: {
        startDayOfWeek: 1,
        dayNames: ['Zondag', 'Maandag', 'Dinsdag', 'Woensdag', 'Donderdag', 'Vrijdag', 'Zaterdag'],
    },
    theme: {
        common: {
            holiday: {
                color: 'black',
            },
        },
        month: {
            holidayExceptThisMonth: {
                color: 'rgba(51, 51, 51, 0.4)',
            },
        },
    }
});

calendar.tuiCalendar.on('afterRenderEvent', (eventObj) => {
    console.log(eventObj);
});

calendar.tuiCalendar.createEvents([
    {
        id: 'event3',
        calendarId: 'cal2',
        title: 'Vacation',
        start: '2022-12-27',
        end: '2022-12-27',
        description: "Deze periode hebben we vacantie",
        isAllday: true,
        category: 'allday',
    },
    {
        id: 'event3',
        calendarId: 'cal2',
        title: 'Vacation',
        start: '2022-12-29',
        end: '2022-12-29',
        description: "Deze periode hebben we vacantie",
        isAllday: true,
        category: 'allday',
    },
]);
```
When i switch ```isReadOnly``` to ```false``` it will output 2 events in the console. When ```isReadOnly``` is ```true``` it will not log anything.

## Expected Behavior

I couldn't find anything in the docs about this event not working in read only mode (some other events in the docs do state ```isReadOnly: false``` is needed). But the events that need ```isReadOnly: false``` are mostly events that only happen when editing/changing/removing things which isn't posible in read only mode so it makes sense. I think there are many use cases where you want the read only mode in combination with this event.

### PR DESCRIPTION
**Pull request by blacktoast**
### Please check if the PR fulfills these requirements

- [x] It's the right issue type on the title
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] It does not introduce a breaking change or has a description of the breaking change

### Description
 - afterRenderEvent works in readonly mode
 - add test in this feature 

Resolves #1327 

---
Thank you for your contribution to TOAST UI product. 🎉 😘 ✨

### CODE REVIEW

**Comment by adhrinae** [comment]
@blacktoast 

Thanks for your contribution 🙌 . I'll review the PR in a few days though it's simple and transparent.

File: apps/calendar/src/components/events/horizontalEvent.spec.tsx [hunk 1]
```diff
@@ -341,4 +366,18 @@ describe('isReadOnly', () => {
...
+    const handler = jest.fn();
+    eventBus.on('afterRenderEvent', handler);
+    const { props } = setup();
+
+    // When (resizingWidth)
+    render(<HorizontalEvent {...props} />, { eventBus });
+
+    // Then
+    expect(handler).toBeCalled();
+  });
```
Comment (adhrinae) [inline apps/calendar/src/components/events/horizontalEvent.spec.tsx:382]: This test should belong to the `Firing 'afterRenderEvent'` block.

https://github.com/nhn/tui.calendar/blob/main/apps/calendar/src/components/events/horizontalEvent.spec.tsx#L77

And though other tests should be changed too, how about changing the assertion to the `toHaveBeenCalledTimes(1)`?

File: apps/calendar/src/components/events/horizontalEvent.tsx [hunk 2]
```diff
@@ -170,7 +170,7 @@ export function HorizontalEvent({
   });
 
   useEffect(() => {
-    if (isDraggableEvent) {
+    if (isNil(resizingWidth) && isNil(movingLeft)) {
```
Comment (adhrinae) [inline apps/calendar/src/components/events/horizontalEvent.tsx:173]: It's better to make this condition a variable.

Let's create a variable above L147(`isDraggableEvent`), then apply other conditions inside this component.

```diff
@@ -16,7 +16,7 @@
 import { DraggingState } from '@src/slices/dnd';
 import { isSameDate } from '@src/time/datetime';
 import { passConditionalProp } from '@src/utils/preact';
-import { isNil } from '@src/utils/type';
+import { isPresent } from '@src/utils/type';
 
 import type { CalendarColor } from '@t/options';
 
@@ -144,8 +144,8 @@
   const eventContainerRef = useRef<HTMLDivElement>(null);
 
   const { isReadOnly, id, calendarId } = uiModel.model;
-  const isDraggableEvent =
-    !isReadOnlyCalendar && !isReadOnly && isNil(resizingWidth) && isNil(movingLeft);
+  const isDraggingGuideEvent = isPresent(resizingWidth) || isPresent(movingLeft);
+  const isDraggableEvent = !isReadOnlyCalendar && !isReadOnly && !isDraggingGuideEvent;
 
   const startDragEvent = (className: string) => {
     setDraggingEventUIModel(uiModel);
@@ -160,8 +160,7 @@
     if (
       draggingState === DraggingState.DRAGGING &&
       draggingEventUIModel?.cid() === uiModel.cid() &&
-      isNil(resizingWidth) &&
-      isNil(movingLeft)
+      !isDraggingGuideEvent
     ) {
       setIsDraggingTarget(true);
     } else {
@@ -170,7 +169,7 @@
   });
 
   useEffect(() => {
-    if (isNil(resizingWidth) && isNil(movingLeft)) {
+    if (isDraggingGuideEvent) {
       eventBus.fire('afterRenderEvent', uiModel.model.toEventObject());
     }
     // This effect is only for the first render.
```

**Review by adhrinae — CHANGES_REQUESTED** [review]

File: apps/calendar/src/components/events/horizontalEvent.spec.tsx (hunk 1 above)
Comment (blacktoast) [reply apps/calendar/src/components/events/horizontalEvent.spec.tsx:382]: To put this code in the 'afterRenderEvent' block. You must create a read-only setup function 
like:
```ts
 function setupReadOnly() {
    const eventBus = new EventBusImpl<ExternalEventTypes>();
    const handler = jest.fn();
    eventBus.on('afterRenderEvent', handler);
    const uiModel = new EventUIModel(
      new EventModel({
        id: '1',
        start: new Date(2020, 0, 1, 10, 0),
        end: new Date(2020, 0, 1, 12, 0),
         isReadOnly: true,
      })
    );
    const props = {
      uiModel,
      eventHeight: 30,
      headerHeight: 0,
    };

    return {
      props,
      eventBus,
      handler,
    };
  }
```
It's okay?

Comment (blacktoast) [reply apps/calendar/src/components/events/horizontalEvent.spec.tsx:382]: it's a good idea to change it to `toHaveBeenCalledTimes(1) `
 I'll change it.

File: apps/calendar/src/components/events/horizontalEvent.tsx (hunk 2 above)
Comment (blacktoast) [reply apps/calendar/src/components/events/horizontalEvent.tsx:173]: It's good, i'll change this code

File: apps/calendar/src/components/events/horizontalEvent.spec.tsx (hunk 1 above)
Comment (adhrinae) [reply apps/calendar/src/components/events/horizontalEvent.spec.tsx:382]: > To put this code in the 'afterRenderEvent' block. I should create a read-only setup function like:
> 
> ```ts
>  function setupReadOnly() {
>     const eventBus = new EventBusImpl<ExternalEventTypes>();
>     const handler = jest.fn();
>     eventBus.on('afterRenderEvent', handler);
>     const uiModel = new EventUIModel(
>       new EventModel({
>         id: '1',
>         start: new Date(2020, 0, 1, 10, 0),
>         end: new Date(2020, 0, 1, 12, 0),
>          isReadOnly: true,
>       })
>     );
>     const props = {
>       uiModel,
>       eventHeight: 30,
>       headerHeight: 0,
>     };
> 
>     return {
>       props,
>       eventBus,
>       handler,
>     };
>   }
> ```
> 
> It's okay?

Hmm.. Not bad idea. But only the new test uses setup function. I would extract it when I have to write another test for this case.

**Comment by adhrinae** [comment]
LGTM.

The failing test is a known issue so I'm merging this PR.

**Pull request merged**